tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Hierarchical.cpp

```diff
@@ -0,0 +1,2383 @@
+
+  // Limit the top 2 coeffs:
+  // Limit top upper neighbor, limit second upper neighbor
+  helper({7.0, 3.0, 2.5, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, -5.0, 1.0e-18, 0.0},
+         {3.0, 2.0, 1.0 * 0.99, 0.5 * 0.99});
+  // Limit top upper neighbor, limit second lower neighbor
+  helper({7.0, 5.0, 2.5, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 1.1, 1.0e-18, 0.0},
+         {3.0, 2.0, 0.9 * 0.99, 0.5 * 0.99});
+  // Limit top upper neighbor, zero second upper neighbor
+  helper({7.0, 1.0, 2.5, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 1.1, 1.0e-18, 0.0},
+         {3.0, 2.0, 0.0, 0.5 * 0.99});
+  // Limit top upper neighbor, zero second lower neighbor
+  helper({7.0, 5.0, 2.5, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 2.1, 1.0e-18, 0.0},
+         {3.0, 2.0, 0.0, 0.5 * 0.99});
+  // Zero top upper neighbor, limit second upper neighbor
+  helper({7.0, 3.0, 1.9, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, -5.0, 1.0e-18, 0.0},
+         {3.0, 2.0, 1.0 * 0.99, 0.0});
+  // Zero top upper neighbor, limit second lower neighbor
+  helper({7.0, 5.0, 1.9, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 1.1, 1.0e-18, 0.0},
+         {3.0, 2.0, 0.9 * 0.99, 0.0});
+  // Zero top upper neighbor, zero second upper neighbor
+  helper({7.0, 1.0, 1.9, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 1.1, 1.0e-18, 0.0},
+         {3.0, 2.0, 0.0, 0.0});
+  // Zero top upper neighbor, zero second lower neighbor
+  helper({7.0, 5.0, 1.9, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 2.1, 1.0e-18, 0.0},
+         {3.0, 2.0, 0.0, 0.0});
+
+  // Limit top lower neighbor, limit second upper neighbor
+  helper({7.0, 3.0, 4.1, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, -5.0, 1.3, 0.0},
+         {3.0, 2.0, 1.0 * 0.99, 0.7 * 0.99});
+  // Limit top lower neighbor, limit second lower neighbor
+  helper({7.0, 5.0, 4.1, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 1.1, 1.3, 0.0},
+         {3.0, 2.0, 0.9 * 0.99, 0.7 * 0.99});
+  // Limit top lower neighbor, zero second upper neighbor
+  helper({7.0, 1.0, 4.1, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 1.1, 1.3, 0.0},
+         {3.0, 2.0, 0.0, 0.7 * 0.99});
+  // Limit top lower neighbor, zero second lower neighbor
+  helper({7.0, 5.0, 4.1, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 2.1, 1.3, 0.0},
+         {3.0, 2.0, 0.0, 0.7 * 0.99});
+  // Zero top lower neighbor, limit second upper neighbor
+  helper({7.0, 3.0, 4.1, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, -5.0, 3.0, 0.0},
+         {3.0, 2.0, 1.0 * 0.99, 0.0});
+  // Zero top lower neighbor, limit second lower neighbor
+  helper({7.0, 5.0, 4.1, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 1.1, 3.0, 0.0},
+         {3.0, 2.0, 0.9 * 0.99, 0.0});
+  // Zero top lower neighbor, zero second upper neighbor
+  helper({7.0, 1.0, 4.1, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 1.1, 3.0, 0.0},
+         {3.0, 2.0, 0.0, 0.0});
+  // Zero top lower neighbor, zero second lower neighbor
+  helper({7.0, 5.0, 4.1, 0.0}, {3.0, 2.0, 2.0, 1.0}, {-2.1, 2.1, 3.0, 0.0},
+         {3.0, 2.0, 0.0, 0.0});
+}
+
+void run() noexcept {
+  for (size_t order = 2; order < 7; ++order) {
+    package_data(order);
+  }
```

Comment:
Testing `package_data` at a whole bunch of different orders is really just retesting our spectral transforms, so I don't think it's necessary. More important, I think, would be testing that the different limiter inputs are correctly orientated, especially in 2 and 3 dimensions.

I suggest: just test this at one order, but make sure to test at least one case per dimension with a non-identity orientation.

Reply:
Agreed, I think in 1d testing the different orders is not bad, but I don't think it's necessary for all dims. I will add tests for the different orientations and higher dims.